src/Interpreters/executeQuery.cpp

```diff
@@ -346,9 +350,38 @@ QueryLogElement logQueryStart(
 
         if (elem.type >= settings.log_queries_min_type && !settings.log_queries_min_query_duration_ms.totalMilliseconds())
         {
-            if (auto query_log = context->getQueryLog())
-                query_log->add(elem);
+            if (!settings.log_query_settings && settings.log_query_settings.changed)
```

Comment:
Assume setting `log_query_settings` was set to `false` via `SET log_query_settings = false` or via config file. If we now run a query which has *no* SETTINGS clause, e.g. a simple `SELECT 1`, we will trigger the logging on l. 354. As a user, I would be confused to see such messages in the server log file.

Ideally, the logging is only done if the query actually has a `SETTINGS` clause. But idk if that is easy to find out here.

Reply:
I think it is better to check for the effective settings, as it's done here. Because the logs are needed for investigations, we want to know any reasons that the query was not logged.